src/bolt/bolt_client.c

```diff
@@ -35,6 +35,7 @@ bolt_client_t *bolt_client_new
 	buffer_new(&client->msg_buf);
 	buffer_new(&client->read_buf);
 	buffer_new(&client->write_buf);
+	buffer_index(&client->read_buf, &client->ws_frame, 0);
```

Comment:
I think renaming `buffer_index` to `buffer_set_index` would be a bit clearer.
Also consider placing the modified argument (`client->ws_frame`) as the first argument to the function.